### PR TITLE
Update notification mechansim for label CI checks

### LINKED ISSUE #88098: [Label Check] Move label check logic from CI to mergebot
Currently, we have [a CI check](https://github.com/pytorch/pytorch/blob/master/.github/workflows/check-labels.yml#L44) that checks to see if a PR has the appropriate labels on it, and fails the check if the right label is missing.

Instead of making this be something that makes the CI appear to have failed, lets make two changes:

1. Modify the CI check to only share a comment reminding the user to add the appropriate CI (without reporting a failing check)
2. Update mergebot rules in https://github.com/pytorch/pytorch/blob/master/.github/scripts/trymerge.py to 1) fail the merge if the PR is missing the required labels and 2) post a comment explaining that the merge was failed due to the missing label

Original customer request:

> Can we make the "needs labels" signal not part of the actual CI signal? I use the GitHub green/red indicator to easily check if a PR succeeded CI or not, but lack of labels turns this into a fail and obscures the signal.
> 
> Instead, **the merge bot should just reject merges if they don't have a label**. The idea being, it's very easy to immediately add the label and try again. You don't need this signal ahead of time!

### PR DESCRIPTION
**Pull request by Goldspear**
Fixes #88098 

What Changed:
1. `Check_labels` CI job no longer fails, but only adding a comment prompt if the required labels is not satisfied.
2. Merge would fail with a comment instead if the required labels are not satisfied.

### CODE REVIEW

File: .github/scripts/trymerge.py [hunk 1]
```diff
@@ -1511,6 +1521,16 @@ def handle_exception(e: Exception, title: str = "Merge failed") -> None:
         gh_post_pr_comment(org, project, args.pr_num, "Cross-repo ghstack merges are not supported", dry_run=args.dry_run)
         return
 
+    if not has_required_labels(pr):
+        gh_post_pr_comment(
+            org,
+            project,
+            args.pr_num,
+            "Can't merge without required labels, see CI job check_labels comment for details",
```
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:1529]: As part of this flow, we'll want to remove the check_labels job as well. 

Can you replace this message with the text `check_labels.py` outputs:
https://github.com/pytorch/pytorch/blob/1facb4a561e89daf44876f98a62f6e9dd0f4332a/.github/scripts/check_labels.py#L21-L26

**Review by ZainRizvi — CHANGES_REQUESTED** [review]
This is looking really good! It just needs a couple more things:

1. Could you please add unit tests in `.github/scripts/test_trymerge.py`
2. We'll also want to remove the check_labels.py and move the corresponding unit tests from `.github/workflows/check-labels.yml` to `test_trymerge.py`
3. Remove the .github/workflows/check-labels.yml workflow

Your call on whether you'd like to do #2 & #3 as part of this PR or as a separate PR

**Comment by Goldspear** [comment]
Thanks @ZainRizvi for the prompt review and suggestions! 

The original issue description suggested to keep the `check_label` CI job, hence the original PR, but I agree it's redundant and should be cleaned up. The new commit now has it removed and the relevant unit tests refractored accordingly. Please kindly take another look when you have a chance :)

File: .github/scripts/trymerge.py (hunk 1 above)
Comment (Goldspear) [reply .github/scripts/trymerge.py]: Removed into `trymerge.py`

**Pull request closed without merging**